src/libm-tester/testerutil.c

```diff
@@ -0,0 +1,230 @@
+  x = flushToZero(x);
+  y = flushToZero(y);
+  if (xisnanf(x) && xisnanf(y)) return 1;
+  if (xisnanf(x) || xisnanf(y)) return 0;
+  if (isinf(x) != isinf(y)) return 0;
+  if (x == POSITIVE_INFINITYf && y == POSITIVE_INFINITYf) return 1;
+  if (x == NEGATIVE_INFINITYf && y == NEGATIVE_INFINITYf) return 1;
+  if (y == 0) {
+    if (isPlusZerof(x) && isPlusZerof(y)) return 1;
+    if (isMinusZerof(x) && isMinusZerof(y)) return 1;
+    return 0;
+  }
+  if (!xisnanf(x) && !xisnanf(y) && !isinf(x) && !isinf(y)) return signf(x) == signf(y);
+  return 0;
+}
+
+int cmpDenormdp(double x, mpfr_t fry) {
+  double y = mpfr_get_d(fry, GMP_RNDN);
+  if (xisnan(x) && xisnan(y)) return 1;
+  if (xisnan(x) || xisnan(y)) return 0;
+  if (isinf(x) != isinf(y)) return 0;
+  if (x == POSITIVE_INFINITY && y == POSITIVE_INFINITY) return 1;
+  if (x == NEGATIVE_INFINITY && y == NEGATIVE_INFINITY) return 1;
+  if (y == 0) {
+    if (isPlusZero(x) && isPlusZero(y)) return 1;
+    if (isMinusZero(x) && isMinusZero(y)) return 1;
+    return 0;
+  }
+  if (!xisnan(x) && !xisnan(y) && !isinf(x) && !isinf(y)) return sign(x) == sign(y);
+  return 0;
+}
+
+int firstTime = 1;
+mpfr_t fra, frb, frc, frd;
+
+double countULPdp(double d, mpfr_t c) {
+  if (firstTime) {
+    mpfr_inits(fra, frb, frc, frd, NULL);
+    firstTime = 0;
+  }
+  double c2 = mpfr_get_d(c, GMP_RNDN);
+  if (c2 == 0 && d != 0) return 10000;
+  if (isnan(c2) && isnan(d)) return 0;
+  if (isnan(c2) || isnan(d)) return 10001;
+  if (c2 == POSITIVE_INFINITY && d == POSITIVE_INFINITY) return 0;
+  if (c2 == NEGATIVE_INFINITY && d == NEGATIVE_INFINITY) return 0;
+
+  double v = 0;
+  if (isinf(d) && !isinfl(mpfr_get_ld(c, GMP_RNDN))) {
+    d = copysign(DBL_MAX, c2);
+    v = 1;
+  }
+
+  //
+  
+  int e;
+  frexpl(mpfr_get_ld(c, GMP_RNDN), &e);
+  mpfr_set_ld(frb, fmaxl(ldexpl(1.0, e-53), DENORMAL_DBL_MIN), GMP_RNDN);
+
+  mpfr_set_d(frd, d, GMP_RNDN);
+  mpfr_sub(fra, frd, c, GMP_RNDN);
+  mpfr_div(fra, fra, frb, GMP_RNDN);
+  double u = fabs(mpfr_get_d(fra, GMP_RNDN));
+
+  return u + v;
+}
+
+double countULP2dp(double d, mpfr_t c) {
+  if (firstTime) {
+    mpfr_inits(fra, frb, frc, frd, NULL);
+    firstTime = 0;
+  }
+  double c2 = mpfr_get_d(c, GMP_RNDN);
+  if (c2 == 0 && d != 0) return 10000;
+  if (isnan(c2) && isnan(d)) return 0;
+  if (isnan(c2) || isnan(d)) return 10001;
+  if (c2 == POSITIVE_INFINITY && d == POSITIVE_INFINITY) return 0;
+  if (c2 == NEGATIVE_INFINITY && d == NEGATIVE_INFINITY) return 0;
+
+  double v = 0;
+  if (isinf(d) && !isinfl(mpfr_get_ld(c, GMP_RNDN))) {
+    d = copysign(DBL_MAX, c2);
+    v = 1;
+  }
+
+  //
+
+  int e;
+  frexpl(mpfr_get_ld(c, GMP_RNDN), &e);
+  mpfr_set_ld(frb, fmaxl(ldexpl(1.0, e-53), DBL_MIN), GMP_RNDN);
+
+  mpfr_set_d(frd, d, GMP_RNDN);
+  mpfr_sub(fra, frd, c, GMP_RNDN);
+  mpfr_div(fra, fra, frb, GMP_RNDN);
+  double u = fabs(mpfr_get_d(fra, GMP_RNDN));
+
+  return u + v;
+}
+
+double countULPsp(float d, mpfr_t c) {
```

Comment:
They differ in just these lines - can we pass the DENORMAL_DBL_MIN and DBI_MIN as a function parameter? Same for the double precision ones.
```
mpfr_set_ld(frb, fmaxl(ldexpl(1.0, e-53), DENORMAL_DBL_MIN), GMP_RNDN);
mpfr_set_ld(frb, fmaxl(ldexpl(1.0, e-53), DBL_MIN), GMP_RNDN);
```